src/components/DataTable/_data-table.scss

```diff
@@ -104,6 +104,24 @@ html[dir='rtl'] {
 }
 
 // Fixes bug when modal is opened, then closed. The toolbar moves to the top
-section.bx--table-toolbar {
+section.#{$prefix}--table-toolbar {
   overflow: unset;
 }
+
+.#{$prefix}--expandable-row .#{$prefix}--table-expand {
+  z-index: 2;
```

Comment:
Would it make sense to use the z-index layer function/helper from Carbon?

https://github.com/carbon-design-system/carbon/blob/master/packages/components/src/globals/scss/_layout.scss#L31

Reply:
That function seems to use drastic changes in z-index. I would hate to have a cell be on top of an overlay for something this small. The only reason we need this is the hack we are using to nudge these cells over the next one.